### DISCUSSION TITLE
Adding an image to $GITHUB_STEP_SUMMARY

### DISCUSSION THREAD

**Discussion by 149kairyu** [community/community#101814] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi community,
I've tried adding an image to the GITHUB_STEP_SUMMARY but all of my attempts have unfortunately been unsuccessful. Is it even possible to add images? If it is, I'm hoping you can help me find what I'm doing wrong.  
__

I wrote a bash script that runs in a GitHub Action step (after the checkout step). The script interacts with other files just fine. The picture is in a folder called .repo_assets, in the same repo as the workflow file, and I'm trying to add it like this:

```
echo "## Heading" >> "$GITHUB_STEP_SUMMARY"
echo "![Image](./.repo_assets/image.png)" >> "$GITHUB_STEP_SUMMARY"
```

Just to make sure that the file is found, I used `if [ -f ./.repo_assets/image.png ]; then` which returns true.

But it ends up looking like this: 
![image](https://github.com/community/community/assets/75082935/1cae3d42-4eb1-497e-8f35-2cc0d91c0ae1)

I tried around a million different paths, adding the image via link, moving the image to the top-level folder, ${{ github.workspace }}, etc., but nothing works. I found [this discussion](https://github.com/orgs/community/discussions/35932) but the suggestions did not work either. I do think that the paths might be the issue but I can't figure out what the correct path is.

FYI, this is on github.com with a self-hosted runner of my company. I contacted our github-responsible team and they suggested different paths, none which worked. Any suggestion is highly appreciated!

**Comment by DarrenChochian** [comment]


**Reply by 149kairyu** [reply]
Thank you for the suggestion! Compying the image to a set path is such a smart way to tackle this issue. Unfortunately, it did not work out either. I attempted your code, $GITHUB_OUTPUT instead of set-output (it seems to be deprecated), and just moving the image in the same step, without the Output. The path might not be a problem after all? I tried it with a different picture but that didn't do the trick either. 
I haven't seen anyone sucessfully add a picture yet, maybe it's not possible after all? Either way, this is taking up way too much time so I'll just give up for now.
Thanks again though! Your help is much appreciated

**Comment by ghost** [comment, marked as answer]
Here's how you can accomplish this:
1. Reference the Image with an Absolute Path: When you use a relative path in the GITHUB_STEP_SUMMARY, it might not resolve correctly because the GitHub Actions runner may not be executing in the directory you expect. Instead, use an absolute path to reference the image.

2. Use the Raw GitHub Content URL: If the image is stored in your repository, you can reference it using the raw content URL from GitHub. This is especially useful if relative paths aren't working.

Here's how you can modify your script:
```
echo "## Heading" >> "$GITHUB_STEP_SUMMARY"

# Replace 'owner/repo', 'branch', and 'path_to_image' with your repository's details
echo "![Image](https://raw.githubusercontent.com/owner/repo/branch/.repo_assets/image.png)" >> "$GITHUB_STEP_SUMMARY"
```
1. Ensure the Image is Accessible: If your repository is private, the raw GitHub content URL method might not work as expected due to access restrictions. In such cases, consider hosting the image externally where it can be publicly accessed, or use a method to authenticate requests to the raw content URL.

2. Check File Existence: It's good that you're checking if the file exists. Make sure the path in the check (if [ -f ./.repo_assets/image.png ]; then) matches the path used in the echo command.

3. Review Runner Environment: Since you're using a self-hosted runner, ensure that the runner's environment has access to the necessary paths and that there are no restrictions on accessing internet resources (in case you're using an external URL for the image).

4. Debugging Tips: To further debug, you might want to add additional steps in your workflow to list the contents of the directory where the image is supposed to be, to confirm that the image is in the expected location relative to the GitHub Actions runner's current working directory.

**Reply by 149kairyu** [reply]
Using an external URL did the trick! It's kind of confusing that this was necessary as the script can interact with other files just fine, but I don't want to complain. Thank you for your support!  
It was just for a bit of whimsy but now Gandalfs appears when someone starts triggers a workflow they're not supposed to:
![image](https://github.com/community/community/assets/75082935/b2293e96-2602-4324-b587-63cb03f0e87b)

For anyone else struggling with this, here's my code:
``` 
echo "![Image](external URL)" >> "$GITHUB_STEP_SUMMARY"
```

replace external URL with the actual URL, e. g. google.com. I wanted to reduce the size of the picture so I ended up using html: 
```
echo '<img src="external URL" alt="Image" width="300">' >> "$GITHUB_STEP_SUMMARY"
```
